crates/deltalake-core/src/kernel/actions/types.rs

```diff
@@ -274,6 +274,26 @@ pub enum WriterFeatures {
     Other(String),
 }
 
+#[allow(clippy::from_over_into)]
+impl Into<usize> for WriterFeatures {
+    fn into(self) -> usize {
+        match self {
+            WriterFeatures::Other(_) => 0,
+            WriterFeatures::AppendOnly | WriterFeatures::Invariants => 2,
+            WriterFeatures::CheckConstraints => 3,
+            WriterFeatures::ChangeDataFeed | WriterFeatures::GeneratedColumns => 4,
+            WriterFeatures::ColumnMapping => 5,
+            WriterFeatures::IdentityColumns
+            | WriterFeatures::DeletionVectors
+            | WriterFeatures::RowTracking
+            | WriterFeatures::TimestampWithoutTimezone
+            | WriterFeatures::DomainMetadata
+            | WriterFeatures::V2Checkpoint
+            | WriterFeatures::IcebergCompatV1 => 7,
+        }
+    }
+}
+
```

Comment:
Are we using this somewhere? As per our discussion on Slack, not entirely sure how and where we would use this.

Reply:
No, I think this was me speculating about what we might need for determining say minimum versions across the entire hashset of features. I don't have any reason to say it should stay so I'm fine removing it if you would like.